src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -114,6 +114,18 @@ internal void InitializeExtendedAttributesWithExisting(IEnumerable<KeyValuePair<
         {
             Debug.Assert(_ea == null);
             _ea = new Dictionary<string, string>(existing);
+            foreach (KeyValuePair<string, string> kvp in _ea)
```

Comment:
That `Dictionary<TKey, TValue>` ctor already loops through `existing` and the foreach will loop again. 
https://github.com/dotnet/runtime/blob/14123c989d6de088ac0ea24bc1d7f1d6b507f06f/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs#L97-L106

It may be worth measure what's the impact of adding the elements in `existing` + validating as you currently have versus validating + adding the elements in one loop.

Or, is there no other place where we loop through `_ea`?